sgx/common/src/ec_key_cert_unmarshal.c

```diff
@@ -0,0 +1,46 @@
+/**
+ * @file ec_key_cert_unmarshal.c
+ *
+ * @brief Provides implementation for converting binary (DER) formatted
+ *        keys and/or certificates into structs used by the OpenSSL API.
+ */
+
+#include "ec_key_cert_unmarshal.h"
+
+/*****************************************************************************
+ * unmarshal_ec_der_to_pkey()
+ ****************************************************************************/
+int unmarshal_ec_der_to_pkey(uint8_t ** ec_der_bytes_in,
```

Comment:
What happens if ```*ec_der_bytes_in``` is ```NULL```? Does ```d2i_PrivateKey``` then return ```NULL``` so we catch it in that call?

Reply:
I don't think d2i_PrivateKey() checks for a NULL pointer being passed in. I didn't look at the OpenSSL source code, but when I tried it (substituted the &buf_in parameter with NULL) it core dumped. I added a check for this.